chain/chain/src/chain.rs

```diff
@@ -391,6 +396,60 @@ pub fn check_known(
     check_known_store(chain, block_hash)
 }
 
+// Cache with the mappint from CryptoHash (blocks, chunks) to the number milliseconds that it took to process them.
+// Used only for debugging purposes.
+pub static CRYPTO_HASH_TIMER_RESULTS: Lazy<Mutex<LruCache<CryptoHash, u64>>> =
+    Lazy::new(|| Mutex::new(LruCache::new(10000)));
+
+/// Struct to measure computation times related to different CryptoHashes (for example chunk or block computations).
+/// It stores the data in the global LRU cache, which allows it to be read afterwards.
+///
+/// Example use:
+/// ```rust, ignore
+/// {
+///     let _timer = CryptoHashTime(block.hash);
+///     // Do stuff with the block
+///     // timer gets released at the end of this block
+/// }
+/// ```
+pub struct CryptoHashTimer {
+    key: CryptoHash,
+    start: Instant,
+}
+
+impl CryptoHashTimer {
+    pub fn new(key: CryptoHash) -> Self {
+        CryptoHashTimer { key, start: Clock::instant() }
+    }
+    pub fn get_timer_value(key: CryptoHash) -> Option<u64> {
+        CRYPTO_HASH_TIMER_RESULTS.lock().unwrap().get(&key).cloned()
+    }
+}
+
+impl Drop for CryptoHashTimer {
+    fn drop(&mut self) {
+        CRYPTO_HASH_TIMER_RESULTS
+            .lock()
+            .unwrap()
+            .put(self.key, (Clock::instant() - self.start).as_millis() as u64);
```

Comment:
Let's extract `(Clock::instant() - self.start).as_millis() as u64` to a variable, so that it's outside of the lock.

Reply:
done